knodle/labeler/CheXpert/matching.py

```diff
@@ -0,0 +1,159 @@
+"""
+This code builds upon the CheXpert labeler from Stanford ML Group.
+It has been slightly modified to be compatible with knodle.
+The original code can be found here: https://github.com/stanfordmlgroup/chexpert-labeler
+
+----------------------------------------------------------------------------------------
+
+Define observation matcher class.
+"""
+import bioc
+import itertools
+import os
+import re
+from collections import defaultdict
+from typing import DefaultDict
+
+from .config import CheXpertConfig
+from .utils import z_matrix_fct, get_rule_idx
+
+
+class Matcher(object):
+    """
+    Find mentions matching observations in report(s).
+
+    Original code:
+    https://github.com/stanfordmlgroup/chexpert-labeler/blob/master/stages/extract.py
+    """
+    def __init__(self, config: CheXpertConfig, chexpert_data: bool = True):
+        self.labeler_config = config
+        self.observation2mention_phrases = self.load_phrases(self.labeler_config.mention_data_dir)
+        self.observation2unmention_phrases = self.load_phrases(self.labeler_config.unmention_data_dir)
+
+        # Add CheXpert specific unmention phrases.
+        if chexpert_data:
+            self.add_unmention_phrases()
+
+    def load_phrases(self, phrases_dir: str) -> DefaultDict:
```

Comment:
FYI: It seems to me that format could be simplified using tables. Then it'd just be a pd.read_csv(). But totally fine!